resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -115,7 +116,7 @@ public void clear() {
     @Override
     public boolean clear(String hostname) {
         checkNotNull(hostname, "hostname");
-        Entries entries = resolveCache.remove(hostname);
+        Entries entries = resolveCache.remove(appendDot(hostname));
```

Comment:
What happens if there is a search path?  If my search path is `netty.io`  then the domain name `foo` would be incorrectly cached as `foo.`  right?

Reply:
We should cache foo.netty.io in this case.

Reply:
@carl-mastrangelo based on your question (thanks!) I also ensured we correctly query the cache for the use-case (+ added a test).